posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -68,6 +68,8 @@ public class PostHog private constructor(
     private var isIdentifiedLoaded: Boolean = false
     private var isPersonProcessingLoaded: Boolean = false
 
+    private lateinit var currentScreen: String
```

Comment:
I think this approach leaves room for too many false positives.
eg I call `screen(x)` for screen X, but I don't call `screen(y)` for screen Y, when events are captured in the screen `y`, the screen name is `x`.

Reply:
I see, you are correct. The problem with using `PostHogActivityLifecycleCallbackIntegration.kt` is that we would have to add method signature in `PostHogInterface`, and i dont know if it make sense.

see theses changes
https://github.com/thisames/posthog-android/commit/7d4a6f5694798f9560d3c629193bf608cd961819

do you think this make sense?

Reply:
I think we should add a new concept to the SDK that allows enriching the event after being captured, for example, an event processor.
I will make an example and you can expand from there

Reply:
See this example https://github.com/PostHog/posthog-android/compare/chore/processors?expand=1

Reply:
Thank you, you made it much easier for me. I will update this PR with the changes and then you check and give me the ok, if it works I will do the unit tests

Reply:
Another option is to use the [super properties](https://posthog.com/docs/libraries/android) for the screen name
https://github.com/PostHog/posthog-android/blob/3a0ede85b9ed25dd993ff3944f74a7187dc175a7/posthog/src/main/java/com/posthog/PostHog.kt#L783-L802
both would work the same

Reply:
I prefer the solution with processors, it's working well